docs/miscellaneous.rst

```diff
@@ -85,6 +88,56 @@ specification
 ABI specification requires arguments to be padded to multiples of 32
 bytes.  The internal function calls use a different convention.
 
+
+.. index: overflow
+
+****************************************
+Internals - Cleaning Up Overflows in EVM
+****************************************
+
+When a value is shorter than 256-bit, sometimes the remaining bits
+must be cleaned.
+The Solidity compiler is designed to clean such remaining bits before any operations
+that might be broken by the garbage in the remaining bits.  For
+example, before writing a value to the memory, the remaining bits need
+to be cleared because the memory contents can be used for computing
+hashes or sent as the data of a message call.  Similarly, before
+storing a value in the storage, the remaining bits need to be cleaned
+because otherwise the garbled value can be observed.
+
+On the other hand, we do not clean the bits if the immediately
+following operation is not affected.  For instance, since any non-zero
+value is considered ``true`` by ``JUMPI`` instruction, we do not clean
+the boolean values before they are used as the condition for
+``JUMPI``.
+
+In addition to the design principle above, the Solidity compiler
+cleans input data when it is loaded onto the stack.
+
+Different types have different rules for cleaning up overflows:
+
++---------------+---------------+------------------+
+|Type           |Cleaned Form   |Overflow Means    |
```

Comment:
This sounds like a condition for overflow, but what we actually want to describe is the procedure. So what about `Action on out-of-range` (to me, overflow is more like the process of overflowing, not reading invalid data).

And then the bool line would be: nonzero is converted to 1.